Microsoft.Toolkit.Uwp.UI.Media/Shadows/AttachedCardShadow.cs

```diff
@@ -48,29 +68,54 @@ public double CornerRadius
             set => SetValue(CornerRadiusProperty, value);
         }
 
+        /// <summary>
+        /// Gets or sets the mode use to clip inner content from the shadow.
+        /// </summary>
+        public InnerContentClipMode InnerContentClipMode
+        {
+            get => (InnerContentClipMode)GetValue(InnerContentClipModeProperty);
+            set => SetValue(InnerContentClipModeProperty, value);
+        }
+
         /// <inheritdoc/>
         public override bool IsSupported => SupportsCompositionVisualSurface;
 
         /// <inheritdoc/>
         protected internal override bool SupportsOnSizeChangedEvent => true;
 
+        /// <inheritdoc/>
+        protected internal override void OnElementContextInitialized(AttachedShadowElementContext context)
+        {
+            UpdateVisualOpacityMask(context);
+            base.OnElementContextInitialized(context);
+        }
+
         /// <inheritdoc/>
         protected override void OnPropertyChanged(AttachedShadowElementContext context, DependencyProperty property, object oldValue, object newValue)
         {
             if (property == CornerRadiusProperty)
             {
+                UpdateShadowClip(context);
+                UpdateVisualOpacityMask(context);
+
                 var geometry = context.GetResource(RoundedRectangleGeometryResourceKey);
                 if (geometry != null)
                 {
                     geometry.CornerRadius = new Vector2((float)(double)newValue);
                 }
-
+            }
+            else if (property == InnerContentClipModeProperty)
+            {
                 UpdateShadowClip(context);
+                UpdateVisualOpacityMask(context);
+                SetElementChildVisual(context);
             }
             else
             {
                 base.OnPropertyChanged(context, property, oldValue, newValue);
             }
+
+            base.OnPropertyChanged(context, property, oldValue, newValue);
```

Comment:
This extra line could cause PropertyChanged to be emitted twice, is that intended?

Reply:
Fixed :)